contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,280 @@
+type MemoryResources struct {  
+  // The time at which these metrics were updated.  
+  Timestamp metav1.Time  
+  // The memory capacity, in bytes  
+  CapacityBytes *uint64
+  // The available memory, in bytes  
+  // This is the number of bytes which are not included in the working set memory
+  // Working set memory includes recently accessed memory, dirty memory, and kernel memory.
+  AvailableBytes *uint64  
+}  
+
+// FilesystemResources contains data about filesystem disk resources.  
+type FilesystemResources struct {  
+  // The time at which these metrics were updated.  
+  Timestamp metav1.Time   
+  // The filesystem name, which uniquely identifies a filesystem  
+  Filesystem string  
+  // AvailableBytes represents the storage space available (bytes) for the filesystem.  
+  AvailableBytes *uint64    
+  // CapacityBytes represents the total capacity (bytes) of the filesystems underlying storage.  
+  CapacityBytes *uint64  
+  // InodesFree represents the free inodes in the filesystem.  
+  InodesFree *uint64  
+  // Inodes represents the total inodes in the filesystem.  
+  Inodes *uint64   
+}  
+
+// PodUsage holds pod-level metrics.  
+type PodUsage struct {  
+  // UID of the pod  
+  PodUID string  
+  // Metrics pertaining to pod total usage of cpu  
+  // This may include additional overhead not included in container usage statistics.  
+  CPU *CpuUsage  
+  // Metrics pertaining to pod total usage of system memory  
+  // This may include additional overhead not included in container usage statistics.  
+  Memory *MemoryUsage   
+  // Metrics of containers in the pod.  
+  Containers []ContainerUsage  
+  // Metrics pertaining to volume usage of filesystem resources.  
+  Volumes []VolumeUsage  
+}  
+
+// ContainerUsage holds container-level usage metrics.  
+type ContainerUsage struct {  
+  // ID of the container  
+  ContainerID string  
+  // Metrics pertaining to container usage of cpu  
+  CPU *CpuUsage  
+  // Metrics pertaining to container usage of system memory  
+  Memory *MemoryUsage  
+  // Metrics pertaining to container rootfs usage of disk.  
+  // Rootfs.UsedBytes is the number of bytes used for the container write layer.  
+  Rootfs *FilesystemUsage   
+  // Metrics pertaining to container logs usage of Disk.  
+  Logs *FilesystemUsage  
+}  
+
+// CpuUsage holds statistics about the amount of cpu time consumed  
+type CpuUsage struct {  
+  // The time at which these Metrics were updated.  
+  Timestamp metav1.Time  
+  // Average CPU usage rate over sample window (across all cores), in "nano cores".  
+  // The "core" unit represents nanoseconds of CPU time consumed per second.  
+  // For example, 5 nanocores means the process averaged 5 nanoseconds 
+  // of cpu time per second during the sample window.
+  UsageRateNanoCores *uint64  
+  // Cumulative CPU usage (sum of all cores) since object creation.  
+  AggregateUsageCoreNanoSeconds *uint64   
```

Comment:
nit: Suggest using `CumulativeUsageNanoSeconds`. The keyword `Core` is confusing.